specifyweb/specify/api_tests.py

```diff
@@ -1234,6 +1234,220 @@ def _get_record_data(pre_merge=False):
         merged_data = fix_record_data(_get_record_data(True), target_model, target_model.name.lower(), new_record_id, old_record_ids)
 
         self.assertDictEqual(merged_data, _get_record_data())
+
+    def test_replace_localities(self):
```

Comment:
add more complicated unit tests, where data comes from the frontend. 

Have at least one unit test that triggers this function to actually fix orderings
https://github.com/specify/specify7/blob/82a3542153ba43c1cdfd4fec4bb60cc7654ceb44/specifyweb/specify/record_merging.py#L93

see https://github.com/specify/specify7/blob/82a3542153ba43c1cdfd4fec4bb60cc7654ceb44/specifyweb/specify/api_tests.py#L1004 where frontend throws bad ordering, but backend is smart enough to fix that (bc of that function).  In the data that frontend sends in the above unit test, check if order numbers are unique. at least one of this kind is necessary.